light_curves/code_src/sample_selection.py

```diff
@@ -378,9 +378,9 @@ def get_sdss_sample(coords, labels, *, num=10, zmin=0, zmax=10, randomize_z=Fals
         res = SDSS.query_sql(query, data_release = 16)
 
         SDSS_coords = [SkyCoord(ra, dec, frame='icrs', unit='deg') for ra, dec in zip(res['ra'], res['dec'])]
-        SDSS_labels = ['SDSS' for ra in res['ra']]
+        SDSS_labels = ['SDSS_QSO' for ra in res['ra']]
 
-        coords.extend(SDSS_coords)
+        coords.extend(zip(SDSS_coords, res['z']))  # add tuple (skycoord, redshift) for each object
```

Comment:
At some point, I lost track that we're still mutating the input lists rather than properly returning a new one? This is a rather unexpected/dangerous behavior and if it's not too late would suggest refactoring.

Reply:
> At some point, I lost track that we're still mutating the input lists rather than properly returning a new one?

Yes, all of the `get_*_sample` functions do this. This PR doesn't change that in any way. I agree that a refactor would would be great, but it has never risen to a high priority for me.

Reply:
yes, I suppose we missed all the functions in `sample_selection` when we did #156 and #171

Reply:
I agree that changing all the get_*_sample functions at this point is not warranted.